test/helpers.test.ts

```diff
@@ -0,0 +1,28 @@
+import {removeParamsFromUrl} from "../src/lib/helpers";
+
+
+test('removeParamsFromUrl() should remove Gotrue params from hash, ignoring case', async () => {
+    expect(removeParamsFromUrl('/some/path#p1=value&p2=value', ['p1', 'p2']))
+        .toBe('/some/path');
+
+    expect(removeParamsFromUrl('/some/path#p1=value&p2=value', ['p1']))
+        .toBe('/some/path&p2=value');
```

Comment:
is this resulting url `/some/path&p2=value` invalid?

Reply:
Aye, nice catch @awalias Looks like the url format is off a bit. The # should not be removed

Reply:
Actually I think that in this case the hash  should be removed, at least it is the behavior that I expect personnaly, because here the hash identifies p1 as a gotruejs param and not a URL hash. I will add a test case with : 
/some/path#somehash#p1=value&p2=value which should become /some/path#somehash&p2=value when we remove p1 param.

Reply:
@tartard  Understandable, but then everything after the `#` is part of the hash string and will never be part of the request sent to the server on browser refresh, but /some/path&p2=value on the other hand suggests that p2 is a query parameter, of which it is not. Essentially, nothing else goes after the hash string

Reply:
As a reminder, my initial case is that I am using a hash based router in my client webapp, and gotrue-js breaks the routing when it removes all the hash. My fix aims to make gotrue-js remove only the parameters it uses in order to be able to set a SITE_URL=http://host/#/auth/ for example in supabase. But since the first gotrue param is always preceded by a '#', gotrue would redirect to http://host/#/auth#access_token=...
That is why I want gotrue-js to remove the '#access_token...' part, including the '#', and not the full ''#/auth/#access_token...' as it is currently doing. 
If I understand you well, I see your point but I think it would not be a problem since everything after the #access_token... is set by gotrue and will be removed by gotrue-js after successfully setting the session. There is no real case where the 
'#p1' would be removed but not the '&p2'. I just made this test case to show the precise behavior of my code.
But I understand that it can be confusing, so maybe I should change my code to not remove params one by one, but rather remove a hash and everything after this hash IF this hash is immediately followed by a gotrue param (like 'access_token'. 
Another solution could be to make gotrue (backend) precede its params with a '&' if a hash is already present in the redirect url.

Reply:
> My fix aims to make gotrue-js remove only the parameters it uses in order to be able to set a SITE_URL=http://host/#/auth/ for example in supabase. But since the first gotrue param is always preceded by a '#', gotrue would redirect to http://host/#/auth#access_token=

This should also be a test case on it's own as it's pretty different from all the available ones... Where there's multiple `#` and I still believe it could be achieved by this PR